ReactiveCocoa/Swift/Property.swift

```diff
@@ -8,66 +8,457 @@ public protocol PropertyType {
+	@warn_unused_result(message="Did you forget to use the composed property?")
+	private func lift<P: PropertyType, U>(transform: SignalProducer<Value, NoError> -> SignalProducer<P.Value, NoError> -> SignalProducer<U, NoError>) -> P -> AnyProperty<U> {
+		return { otherProperty in
+			return AnyProperty(transformingFirst: self, second: otherProperty, using: transform)
+		}
+	}
+
+	/// Maps the current value and all subsequent values to a new property.
+	@warn_unused_result(message="Did you forget to use the composed property?")
+	public func map<U>(transform: Value -> U) -> AnyProperty<U> {
+		return lift { $0.map(transform) }
+	}
+
+	/// Combines the current value and the subsequent values of two `Property`s in
+	/// the manner described by `Signal.combineLatestWith:`.
+	@warn_unused_result(message="Did you forget to use the composed property?")
+	public func combineLatest<P: PropertyType>(with other: P) -> AnyProperty<(Value, P.Value)> {
+		return lift(SignalProducer.combineLatestWith)(other)
+	}
+
+	/// Zips the current value and the subsequent values of two `Property`s in
+	/// the manner described by `Signal.zipWith`.
+	@warn_unused_result(message="Did you forget to use the composed property?")
+	public func zip<P: PropertyType>(with other: P) -> AnyProperty<(Value, P.Value)> {
+		return lift(SignalProducer.zipWith)(other)
+	}
+
+	/// Forwards events from `self` with history: values of the returned property
+	/// are a tuple whose first member is the previous value and whose second member
+	/// is the current value. `initial` is supplied as the first member of the first
+	/// tuple.
+	@warn_unused_result(message="Did you forget to use the composed property?")
+	public func combinePrevious(initial: Value) -> AnyProperty<(Value, Value)> {
+		return lift { $0.combinePrevious(initial) }
+	}
+
+	/// Forwards only those values from `self` which do not pass `isRepeat` with
+	/// respect to the previous value. The first value is always forwarded.
+	@warn_unused_result(message="Did you forget to use the composed property?")
+	public func skipRepeats(isRepeat: (Value, Value) -> Bool) -> AnyProperty<Value> {
+		return lift { $0.skipRepeats(isRepeat) }
+	}
+}
+
+extension PropertyType where Value: Equatable {
+	/// Forwards only those values from `self` which is not equal to the previous
+	/// value. The first value is always forwarded.
+	@warn_unused_result(message="Did you forget to use the composed property?")
+	public func skipRepeats() -> AnyProperty<Value> {
+		return lift { $0.skipRepeats() }
+	}
+}
+
+public enum PropertyFlattenStrategy {
+	case Latest
+}
+
+extension PropertyType where Value: PropertyType {
+	/// Returns a property that forwards values from the latest property hold by
+	/// `self`, ignoring values sent on previous inner properties.
+	@warn_unused_result(message="Did you forget to use the composed property?")
+	public func flatten(strategy: PropertyFlattenStrategy) -> AnyProperty<Value.Value> {
```

Comment:
I think `Concat` would actually make sense—as much sense as doing any other flatten operation with `PropertyType`s. Let's either use the normal `FlattenStrategy` or remove `flatten` altogether.